src/msg/async/AsyncConnection.cc

```diff
@@ -821,6 +816,8 @@ void AsyncConnection::process()
     }
   } while (prev_state != state);
 
+  if (need_dispatch_writer)
```

Comment:
I think we need to change to "if (need_dispatch_writer && can_write == WriteStatus::CANWRITE)"

Otherwise, connection may in STATE_CLOSED but we will dispatch write_handler which will result in EventCenter call a deconstructed method